app/controllers/queue_items_controller.rb

```diff
@@ -14,10 +14,32 @@ def create
   def destroy
     queue_item = QueueItem.find(params[:id])
     queue_item.destroy if current_user.queue_items.include?(queue_item)
+    current_user.normalize_queue_item_positions
     redirect_to my_queue_path
   end
 
+  def update_queue
+    begin
+      update_queue_items
+      current_user.normalize_queue_item_positions
+    rescue ActiveRecord::RecordInvalid
+      flash[:danger] = "Invalid position numbers"
+    end
+    redirect_to my_queue_path 
+  end
+
   private
+
+  def update_queue_items
+    ActiveRecord::Base.transaction do 
+      params[:queue_items].each do |queue_item_data|
+        queue_item = QueueItem.find(queue_item_data["id"])
+        if queue_item.user == current_user
+          queue_item.update_attributes!(position: queue_item_data["position"], rating: queue_item_data["rating"])
```

Comment:
Rails 4 introduced `.update` or `.update!` method that does the same as `.update_attributes` or the bang version. Consider using that - it also saves some room :)